src/Apps/Order/Routes/Payment/index.tsx

```diff
@@ -339,13 +340,37 @@ export const PaymentRoute: FC<PaymentRouteProps> = props => {
         selectedPaymentMethod === "SEPA_DEBIT")
     ) {
       handlePaymentStepComplete()
+    } else if (paymentSetupErrorCode && !isPaymentSetupSuccessful) {
+      setIsSavingPayment(false)
+      props.router.push(`/orders/${props.order.internalID}/payment`)
```

Comment:
I did this to reset the form when an error occurs (otherwise the loading screen was being shown indefinitely). There must be a better way...

Reply:
I wonder if a single `resetForm` helper makes sense here

Reply:
That did the trick!! Thank you.